nnstreamer/tensor_filter/tensor_filter_nntrainer.cc

```diff
@@ -126,17 +121,12 @@ int NNTrainerInference::run(const GstTensorMemory *input,
     return -3;
   }
 
-  GstTensorMemory *output_mem = output;
-  for (auto &o : outputs) {
-    if (o == nullptr) {
+  for (size_t idx = 0; idx < output_dims.size(); idx ++) {
+    if (outputs[idx] == nullptr) {
       return -1;
     }
 
-    out = std::const_pointer_cast<nntrainer::Tensor>(o);
-    output_mem->data = out->getData();
-
-    outputTensorMap.insert(std::make_pair(output_mem->data, out));
```

Comment:
With the current implementation, it is possible that the output memory is freed by nntrainer before the nnstreamer can finish its usage. This will be properly fixed with #1440.